packages/runtime/datastore-definitions/src/jsonable.ts

```diff
@@ -3,19 +3,50 @@
  * Licensed under the MIT License.
  */
 
+/**
+ * Type constraint for types that are serializable as JSON or have a custom
+ * alternate type.
+ *
+ * @remarks
+ * Use JsonableTypeWith<never> for just JSON serializable types.
+ */
+export type JsonableTypeWith<T> =
```

Comment:
All the new exports probably want to be `@internal` to go along with what @anthony-murphy is doing as part of cleaning up our public API surface.

Reply:
api-extractor will not let new exports be @internal if they are used by a non-internal type as this is.